### PR TITLE
Added Atbash cipher exercise

### PR DESCRIPTION
**Pull request by icyrockcom**
Let me know if something can be improved.

### CODE REVIEW

File: exercises/atbash-cipher/examples/src/AtbashCipher.purs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+encMap :: Map Char Char
+encMap = fromFoldable (zip plain cipher)
+
+decode :: String -> Maybe String
+decode s = fromCharArray <$> dec (toCharArray s)
+  where dec = filter ((/=) ' ')
+              >>> map (flip lookup decMap)
+              >>> sequence
+
+groupBy :: forall a. Int -> Array a -> Array (Array a)
```
Comment (paf31) [inline exercises/atbash-cipher/examples/src/AtbashCipher.purs]: This is usually called `chunks` or `chunksOf`.

File: exercises/atbash-cipher/examples/src/AtbashCipher.purs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+groupBy n [] = []
+groupBy n a = take n a : groupBy n (drop n a)
+
+encode :: String -> Maybe String
+encode s = group <$> enc (toLower >>> toCharArray $ s)
+  where enc = filter (not <<< (isPunctuation || isSpace))
+              >>> map (flip lookup encMap)
```
Comment (paf31) [inline exercises/atbash-cipher/examples/src/AtbashCipher.purs]: Nitpick: you could use ``(_ `lookup` encMap)`` here.

File: exercises/atbash-cipher/examples/src/AtbashCipher.purs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+              >>> sequence
```
Comment (paf31) [inline exercises/atbash-cipher/examples/src/AtbashCipher.purs]: Use `traverse f` instead of `map f >>> sequence`.

**Comment by icyrockcom** [comment]
Thanks @paf31 for the useful suggestions, really appreciated! I've updated the above + a couple of uses of `>>>` under `encode` as per your earlier comments.

**Comment by paf31** [comment]
Thanks!

**Pull request merged**